spacetemplate/template.go

```diff
@@ -0,0 +1,107 @@
+	"time"
+
+	"github.com/almighty/almighty-core/convert"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+const APIStringType = "spacetemplates"
+
+// SpaceTemplate describes a single space template
+type SpaceTemplate struct {
+	gormsupport.Lifecycle
+	ID          uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	Version     int
+	Name        string
+	Description *string
+	Template    string // This is the actual unmodified YAML template as specified by the user
+}
+
+// GetETagData returns the field values to use to generate the ETag
+func (st SpaceTemplate) GetETagData() []interface{} {
+	return []interface{}{st.ID, st.Version}
+}
+
+// GetLastModified returns the last modification time
+func (st SpaceTemplate) GetLastModified() time.Time {
+	return st.UpdatedAt.Truncate(time.Second)
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (st *SpaceTemplate) TableName() string {
+	return "space_templates"
+}
+
+// Ensure SpaceTemplate implements the Equaler interface
+var _ convert.Equaler = SpaceTemplate{}
+var _ convert.Equaler = (*SpaceTemplate)(nil)
+
+// Equal returns true if two Template objects are equal; otherwise false is returned.
+func (st SpaceTemplate) Equal(u convert.Equaler) bool {
+	other, ok := u.(SpaceTemplate)
+	if !ok {
+		return false
+	}
+	if !st.Lifecycle.Equal(other.Lifecycle) {
+		return false
+	}
+	if !uuid.Equal(st.ID, other.ID) {
+		return false
+	}
+	if st.Version != other.Version {
+		return false
+	}
+	if st.Name != other.Name {
+		return false
+	}
+	if st.Template != other.Template {
+		return false
+	}
+	if st.Description == nil && other.Description == nil {
+		return true
+	}
+	if st.Description != nil && other.Description != nil {
+		return *st.Description == *other.Description
+	}
+	return false
+}
+
+// Validate returns nil if all required fields are set and the parsed template
+// (if any) is valid.
+func (st SpaceTemplate) Validate() error {
+	if st.Name == "" {
+		return errors.NewBadParameterError("name", st.Name).Expected("not empty")
+	}
+	if st.Template == "" {
+		return errors.NewBadParameterError("template", st.Template).Expected("not empty")
+	}
+	parsedTempl, err := st.Parse()
+	if err != nil {
+		return errors.NewBadParameterError("template", st.Template).Expected("parsable template")
+	}
+	if err := parsedTempl.Validate(); err != nil {
+		return errs.Wrapf(err, "parsed template is not valid")
+	}
+	return nil
+}
+
+// Parse parses the YAML Template and returns a parsed representation of it
+func (st SpaceTemplate) Parse() (*ParsedTemplate, error) {
+	parsedTempl, err := ParseStringToTemplate(st.Template)
+	if err != nil {
+		return nil, errs.Wrapf(err, "failed to parse space template YAML: %+v", st.Template)
+	}
+	return parsedTempl, nil
+}
+
+func GetValidEmptyTemplate() string {
+	return `
```

Comment:
The fact of having this template as an empty one is a bit weird. But this is one assessment